src/parser/FileParser.kt

```diff
@@ -0,0 +1,108 @@
+package parser
+
+import CoffeeShop
+import Coordinate
+import java.io.File
+
+/**
+ * Defines a parser.FileParser object, containing different operations on the content of a given .csv file
+ *
+ * created by AlexandraV on 12/25/2018
+ */
+data class FileParser(var fileName: String, var userCoordinate: Coordinate) {
+    val coffeeShopsList: MutableList<CoffeeShop> = mutableListOf()
+    private val lines: List<String>? = File(fileName).useLines { it.toList() } // the reader is closed automatically
+
+    /**
+     *  parses the lines of the .csv file and converts them into CoffeeShops objects, if possible
+     *  All the lines that cannot be converted to CoffeeShops objects will be written into  invalid_coffee_shops.txt file
+     */
+    fun parseCoffeeShopsList(n: Int) {
```

Comment:
This method dose so much more that the name suggests:
- it will parse file
- validate file
- calculate results
- print the results and errors

Let's talk about how to better express this flow in code, do you have some thoughts?